contracts/modules/LoanSettings/LoanSettings.sol

```diff
@@ -91,178 +68,111 @@ contract LoanSettings is State, InterestHandler, LoanSettingsEvents, PausableGua
                 loanParamsLocal.maintenanceMargin,
                 loanParamsLocal.maxLoanTerm
             );
-            emit LoanParamsIdDisabled(
-                loanParamsLocal.id,
-                loanParamsLocal.owner
-            );
+            emit LoanParamsIdDisabled(loanParamsLocal.id, loanParamsLocal.owner);
         }
     }
 
-    function getLoanParams(
-        bytes32[] memory loanParamsIdList)
-        public
-        view
-        returns (LoanParams[] memory loanParamsList)
-    {
-        loanParamsList = new LoanParams[](loanParamsIdList.length);
-        uint256 itemCount;
-
-        for (uint256 i = 0; i < loanParamsIdList.length; i++) {
-            LoanParams memory loanParamsLocal = loanParams[loanParamsIdList[i]];
-            if (loanParamsLocal.id == 0) {
-                continue;
-            }
-            loanParamsList[itemCount] = loanParamsLocal;
-            itemCount++;
-        }
-
-        if (itemCount < loanParamsList.length) {
-            assembly {
-                mstore(loanParamsList, itemCount)
-            }
+    function modifyLoanParams(LoanParams[] calldata loanParamsList) external onlyGuardian {
```

Comment:
I am unsure where the new LoanParams is being written to storage within this function. Also, the require statement ordering seems off with the ternary used within generateLoanParamId as the false includes && that ensures the minimum margin is greater than maintenance which should be placed outside of that function and within the main require statement.

Reply:
fixed